### ISSUE TITLE
Aggregate by fund: handle multifund project

### ISSUE THREAD

**Issue by madewild** [ec-doris/kohesio-backend#73]
Some projects are multifunds and we don't know the breakdown by fund.
@roberto-musmeci can you provide examples?
How do we handle them in the fund aggregates? Show as "other funds" or separate categaory "multifunds"?

**Comment by madewild** [comment]
@roberto-musmeci we are still missing some examples... I propose to postpone this issue.

**Comment by madewild** [comment]
@roberto-musmeci any input? :)

**Comment by madewild** [comment]
We could need some examples of such projects...

**Comment by roberto-musmeci** [comment]
So the rational is the following. 
The CCI Code is composed by 
YEAR + CC(country code)+  DG(lead DG) + FF(fund) + TP(type of program) + SEQ(sequential number)
eg. 2014 BE 16 RF OP 001
 
Therefore a CCI code contains "M" at character position 9 it means that program is multi-fund. The number following the "M" specified which funds fianances that Program (see table below). 

Sometimes, the list of operations break down the project contribution by fund. Sometimes, they don't. 
Knowing that, we can query projects belonging to a Multi-fund Program for which the fund is not specified ([query](https://query.linkedopendata.eu/#SELECT%20%3Fcci%20%3Fp%0AWHERE%20%7B%0A%20%20%20%3Fp%20wdt%3AP35%20wd%3AQ9934%3B%0A%20%20%20%20%20%20wdt%3AP1368%20%3Fprg.%0A%20%20OPTIONAL%7B%3Fprg%20wdt%3AP1367%20%3Fcci%7D%20%0A%20%20%0A%20%20%20FILTER%20NOT%20EXISTS%20%7B%0A%20%20%20%20%20%20%3Fp%20wdt%3AP1584%20%3Fx%0A%20%20%20%7D%0A%20%7D%20%0ALIMIT%205))
Here a relevant example https://dev.kohesio.eu/projects/Q3068676 

The query misses a passage to restrict only to CCI containing M at position 9. I would do a turnourn in python for the moment but I would be curious to learn how to do it on SPARQL if you know 😄  
 
</html>

M1 | ERDF+CF
-- | --
M2 | ESF+ERDF
M3 | ERDF+CF+ESF
M4 | ERDF+IPA
M5 | ERDF+ENI
M6 | ERDF+IPA + ENI
M7 | IPA + ENI
M8 | ERDF+CF+ESF+EMFF+EAFRD (for Partnership Agreement)
M9 | ESF+YEI
M0 | ERDF+ESF+YEI
MA | ERDF+CF+ESF+YEI

</html>

[CCI Code Definition (1).docx](https://github.com/ec-doris/kohesio-backend/files/7959132/CCI.Code.Definition.1.docx)

**Comment by madewild** [comment]
For M3 programmes like https://linkedopendata.eu/wiki/Item:Q2463136, the 3 funds are already encoded at programme-level so that's a start. They were just not materialized at project level.

**Comment by madewild** [comment]
Currently we have these CCIs concerned:

id | CCI
-- | --
1 | 2014PL16M1OP001
2 | 2014PL16CFTA001
3 | 2014EE16M3OP001
4 | 2014SK05M0OP001
5 | 2014SK16RFTA001
6 | 2014FR05SFOP001
7 | 2014FR05M9OP001
8 | 2014FR16M0OP001
9 | 2014FR05M0OP001
10 | 2014FR16M2OP012
11 | 2014FR05SFOP005
12 | 2014FR05M2OP001
13 | 2014FR05SFOP003
14 | 2014FR05SFOP004
15 | 2014FR16M0OP005
16 | 2014FR16M2TA001
17 | 2014HU16M1OP003
18 | 2014HU16M1OP001
19 | 2014HU05M3OP001
20 | 2014NL16RFOP001
21 | 2014TC16I5CB008
22 | 2014IT05M9OP001

We can easily filter out those with no "M" in position 9...

**Comment by madewild** [comment]
Better with some stats about the number of projects involved:

id | CCI | projects
-- | -- | --
1 | 2014FR05SFOP001 | 7093
2 | 2014IT05M9OP001 | 2375
3 | 2014HU16M1OP001 | 1401
4 | 2014FR05M9OP001 | 617
5 | 2014FR16M0OP005 | 557
6 | 2014PL16M1OP001 | 319
7 | 2014FR05SFOP005 | 252
8 | 2014FR05M2OP001 | 165
9 | 2014FR16M2TA001 | 102
10 | 2014FR05SFOP003 | 95
11 | 2014HU16M1OP003 | 91
12 | 2014FR05SFOP004 | 83
13 | 2014FR16M0OP001 | 78
14 | 2014PL16CFTA001 | 76
15 | 2014FR16M2OP012 | 51
16 | 2014SK16RFTA001 | 30
17 | 2014EE16M3OP001 | 23
18 | 2014HU05M3OP001 | 16
19 | 2014NL16RFOP001 | 12
20 | 2014SK05M0OP001 | 4
21 | 2014TC16I5CB008 | 2
22 | 2014FR05M0OP001 | 1

**Comment by madewild** [comment]
Adapted query for reference:

```sparql
PREFIX wd: <https://linkedopendata.eu/entity/>
PREFIX wdt: <https://linkedopendata.eu/prop/direct/>
SELECT ?cci (count(?p) as ?count)
WHERE {
   ?p wdt:P35 wd:Q9934;
      wdt:P1368 ?prg.
  OPTIONAL{?prg wdt:P1367 ?cci} 
  
   FILTER NOT EXISTS {
      ?p wdt:P1584 ?x
   }
} group by ?cci order by desc(?count)
```

Run on https://dev.qa.linkedopendata.eu/user/endpoint?kb=eu&user=commission to avoid timeout

**Comment by D063520** [comment]
Can you concretizize the discussion into a concrete task?  What exaclty should we do? Basically this:

ERDF-ESF - 1000 EUR.

?

**Comment by madewild** [comment]
I think we should:
1. Materialize the funds at project level whenever available at programme level
2. Adapt the backend to allow multiple funds rather than just one
3. Find an acceptable way to display this in the frontend
I can coordinate this.

**Comment by madewild** [comment]
Just to clarify, on the project detail page (e.g. https://dev.kohesio.eu/projects/Q3068676) we currently have:

Fund: n/a
[...]
Total budget: 195 776,93 €
Funded by EU: 166 410,40 € (85%)

It's not planned to have a breakdown by fund like for the beneficiaries... So the scope would be to just list the multiple funds under Fund instead of n/a, right? I think there is a confusion with the beneficiary detail page...

**Comment by D063520** [comment]
So basically the idea is on the project page to infer the fund from the program when it is not directly available. Is this right?

**Comment by madewild** [comment]
I know, what I say is: for most projets the fund is directly available on the project, but for some we have only the programme and we have to jump to the programme to know the fund. So yes we could infer the fund by making the extra jump, but since most projects have already this info why not instead materialize the fund info (deduced from the programme) for the ~ 10.000 projects above? I can handle that.

**Comment by madewild** [comment]
Or: we remove the funds from all projects, keep them only at programme level and change the backend to look always at programme level (but if we did otherwise I guess it was for performance reasons?)

**Comment by D063520** [comment]
wait wait ..... if a project is in a program in does not mean that it is in the funds of the program. If there is only one fund then yes. But for multi-fund programs, a project is generally in one of the funds. Not in all. So if it is not declared by the managing authority, we cannot infer that knowledge in most of the cases .....

**Comment by madewild** [comment]
OK, so when the fund is on the project, it means it was provided by the MA and so it takes precedence on the fund(s) mentioned in the programme. Then yes maybe we can fallback on the programme only when we don't have explicit info at project level.

**Comment by madewild** [comment]
I discussed with @roberto-musmeci and for him we can materialize the info from the programme if not available directly.
So for https://linkedopendata.eu/wiki/Item:Q3068676 instead of having no funds we could materialize the 3 funds available on the programme https://linkedopendata.eu/wiki/Item:Q2463136

I propose to:
1. tackle the projects that are in an RF/SF/CF programme where the fund info is unambiguous and should be added
2. tackle the projects with a M# code
3. talk with @DiaZork to adapt the fundLabel in the backend (we can probably concatenate there to avoid changing the frontend)

**Comment by D063520** [comment]
@roberto-musmeci: are you sure? I remember Thierry was against this ..... it breaks the policy that we try not to change the original data ..... we cannot revert this also, i.e. it will not be easy to remove the once we inferred. This will be written in the history, but we cannot query this information via SPARQL. So once it is done, there is no way back .....

**Comment by roberto-musmeci** [comment]
Thanks @D063520. Let me discuss it with Anne. Maybe I am missing the point

**Comment by D063520** [comment]
The main problem is that a multi-fund program is mixing funds (fund A,B,C). But without the managing authority telling, we do not know from the project to which fund it is belonging (fund A or B or C or several). I'm not 100% sure, but I think it is like this.

**Comment by madewild** [comment]
OK I will already work on the single-funds programmes in the meantime

**Comment by madewild** [comment]
@D063520: we could add a qualifier "method of determination" for the ones deduced from the programmes, in that way it would be easier to revert if needed...

**Comment by D063520** [comment]
yes true, this is an option

**Comment by madewild** [comment]
Here's an update:

id | CCI | projects
-- | -- | --
1 | 2014IT05M9OP001 | 20181
2 | 2014FR05SFOP001 | 3009
3 | 2014HU16M1OP001 | 1401
4 | 2014FR05M9OP001 | 617
5 | 2014FR16M0OP005 | 557
6 | 2014PL16M1OP001 | 319
7 | 2014FR05M2OP001 | 165
8 | 2014FR16M2TA001 | 102
9 | 2014HU16M1OP003 | 91
10 | 2014FR16M0OP001 | 78
11 | 2014FR16M2OP012 | 51
12 | 2014EE16M3OP001 | 23
13 | 2014HU05M3OP001 | 16
14 | 2014SK05M0OP001 | 4
15 | 2014TC16I5CB008 | 2
16 | 2014FR05M0OP001 | 1

2014IT05M9OP001 got much worse (from 2000 to 20000) because @D063520 imported new IT projects
2014FR05SFOP001 is still being processed

Now we are left with the M# programmes
So we go ahead and use determination method? https://linkedopendata.eu/wiki/Property:P29
@roberto-musmeci waiting for your green light

**Comment by roberto-musmeci** [comment]
For the M#, Anne also prefers leaving the mention "n/a" exactly for what @D063520 mentioned [before](https://github.com/ec-doris/kohesio-backend/issues/73#issuecomment-1025668763). We do not take the risk of modifying the data. I thought that from a communciation perspective would not change too much but I understand and agree with you. 
So thanks a lot Dennis for stressing this point.

I would say we can close it once the [mono-funds](https://github.com/ec-doris/kohesio-backend/issues/73#issuecomment-1025813951) are fixed

**Comment by madewild** [comment]
OK perfect. And the two I5 (IPA) projects https://linkedopendata.eu/wiki/Item:Q4038552 and https://linkedopendata.eu/wiki/Item:Q4038554 should be filtered out by @DiaZork  and no be instances of Kohesio projects anymore, right?

**Comment by roberto-musmeci** [comment]
Exactly ! I will provide a list of all OPs in scope of Kohesio

**Comment by madewild** [comment]
Done, now we have only M# funds remaining:

id | CCI | projects
-- | -- | --
1 | 2014IT05M9OP001 | 20181
2 | 2014HU16M1OP001 | 1401
3 | 2014FR05M9OP001 | 617
4 | 2014FR16M0OP005 | 557
5 | 2014PL16M1OP001 | 319
6 | 2014FR05M2OP001 | 165
7 | 2014FR16M2TA001 | 102
8 | 2014HU16M1OP003 | 91
9 | 2014FR16M0OP001 | 78
10 | 2014FR16M2OP012 | 51
11 | 2014EE16M3OP001 | 23
12 | 2014HU05M3OP001 | 16
13 | 2014SK05M0OP001 | 4
14 | 2014FR05M0OP001 | 1
